### PR TITLE
OASIS StyleGAN2 - Adam Fittler s4696807 

### PR DESCRIPTION
**Pull request by adamFittlerUQ**
An implementation of the StyleGAN2 model used on the OASIS brain dataset to generate high resolution stylised brain images.

The files added are broken down below:
- `README.md`: Report and break down of the model, how to use it and display of loss and generated images
- `dataset.py`: The data loader used to load in the OASIS dataset and performs preprocessing on the data
- `modules.py`: The StyleGAN2 model architecture and all of the block and layers required for it. 
- `predict.py`: Loads the saved generator models and inputs style vectors of different noise values to show stylised images
- `train.py`: Training logic for the StyleGAN2 model it saves the generator and discriminator models every 20 epochs 

Please let me know if there are any changed that need to me made :)

### CODE REVIEW

**Comment by adamFittlerUQ** [comment]
LGTM

**Comment by WeiDaiwow** [comment]
# PR marking

# Document

- Title: 
	Good
- Address probelm:
	Good 
- Working Principle:
	Good
- Figure visualization: 
	Good
- dependecy: 
	Missing
- inputs and outputs description:
	Good
- training plot:
	Good
- readme using markdown:
	Good
- dataset preprocessing description:
	Missing
- dataset split description:
	Good

Minor tips (nothing to do with marking)
1. Noise B direction seems wrong in generator figure
2. Equilised layers normally start with normalizing the weights of operation by scaling with certain factors. Equilised learning rate is the result not the operation itself. Weight are initilized to a 0 mean and std 1, not from such distribution
3. The training loss plot does not show the "improvement" in terms of image quality. Since the adversarial training is all about training balance between G and D, the gtraining loss plot only tell us whether two models are still competing with each other. Visuailization is still the best way to tell whether your model is getting better at generating high quality image. Or extra metrics may be needed 

# Check the files:
- modules.py
	Good
- dataset.py
	Good
- train.py
	Good
- predict.py
	Good
	

Minor tips (nothing to do with marking):
1. It is a better practice to zero_grad() the optimzier, not the model itself. While you only have 1 optimizer for each model and zero_grad() the model wont cause any error, it is still better practice. 


# Overall:

- solve the problem:
	Yes
- TF/torch implementation:
	Yes 
- Design quality:
	Great
- code comments:
	Good
- Difficulty: 
	Hard

**Comment by adamFittlerUQ** [comment]
Thanks for the feedback, I will implement it ASAP!!

**Comment by adamFittlerUQ** [comment]
Seems like I accidently updated some code that was not intended I will fix that up ASAP

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks missing -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
No Data leakage
Difficulty: Hard
## Commit Log
Meaningful commit messages
Progressive commits used
## Documentation
ReadMe acceptable, some dependencies etc. missing -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description good

**Comment by shakes76** [comment]
Can't merge, please remove gitignore directory and revert changes to repo README. This will not affect your grade, but cant' merge into main repo until sorted.

**Comment by adamFittlerUQ** [comment]
Will update now, so just checking remove gitignore and revert the previous change?

**Comment by wangzhaomxy** [comment]
Feedback attempt and no feedback marks lost.
